### PR TITLE
Fix detach&attach not to create new COSEMessage instance

### PR DESCRIPTION
**Pull request by kentakayama**

### CODE REVIEW

**Comment by kentakayama** [comment]
I noticed that dumps() method include the payload even after detach_payload() is done.
Let me give some time and fix it.

File: cwt/cose_message.py [hunk 1]
```diff
@@ -309,16 +309,19 @@ def detach_payload(self: Self) -> Tuple[COSEMessage, bytes]:
...
         Returns:
-            Tuple[COSEMessage, bytes]: A byte string of the encoded COSE or a
-                cbor2.CBORTag object, and a byte string of the detached payload.
+            Tuple[COSEMessage, bytes]: The COSE message (self),
+            and a byte string of the detached payload.
         Raises:
             ValueError: The payload does not exist.
         """
 
         if not isinstance(self._payload, bytes):
```
Comment (dajiaji) [inline cwt/cose_message.py:318]: For now, if you want to implement holding the detached payload in a COSEMessage structure:

```suggestion
    if self._msg[2] is None:
```

File: cwt/cose_message.py [hunk 2]
```diff
@@ -309,16 +309,19 @@ def detach_payload(self: Self) -> Tuple[COSEMessage, bytes]:
...
             raise ValueError("The payload does not exist.")
 
-        return COSEMessage(self._type, [self._msg[0], self._msg[1], None, *self._msg[3:]]), self._payload
+        payload = self._payload
+        self._payload = None
+
+        return self, payload
```
Comment (dajiaji) [inline cwt/cose_message.py:324]: For now, if you want to implement holding the detached payload in a COSEMessage structure:

```suggestion
        self._msg[2] = None

        return self, self._payload
```

File: cwt/cose_message.py [hunk 3]
```diff
@@ -334,5 +337,6 @@ def attach_payload(self: Self, payload: bytes) -> COSEMessage:
 
         if self._payload is not None:
```
Comment (dajiaji) [inline cwt/cose_message.py:338]: For now, if you want to implement holding the detached payload in a COSEMessage structure:

```suggestion
        if self._msg[2] is not None:
```

**Comment by kentakayama** [comment]
@dajiaji 
Thank you for your review, now it works.
I have to note that

## COSEMessage.loads() on detached content message works fine
This code below prints like this, and it outputs what I expected.
> 18([<< {1: -7, 3: 0} >>, {4: << -18, -18 >>}, null, h'6520BBAF2081D7E0ED0F95F76EB0733D667005F7467CEC4B87B9381A6BA1EDE8E00DF29F32A37230F39A842A54821FDD223092819D7728EFB9D3A0080B75380B'])
``` python
"""
import sys
import subprocess
from cwt import COSEMessage
18([
  / protected: / h'A201260300',
  / unprotected: / {4: h'3131'},
  / payload: / null / detached /,
  / signature: / h'6520BBAF2081D7E0ED0F95F76EB0733D667005F7467CEC4B87B9381A6BA1EDE8E00DF29F32A37230F39A842A54821FDD223092819D7728EFB9D3A0080B75380B'
])
"""
expected_detached_cose_message = COSEMessage.loads(
    bytes.fromhex(
        "D28445A201260300A104423131F658406520BBAF2081D7E0ED0F95F76EB0733D667005F7467CEC4B87B9381A6BA1EDE8E00DF29F32A37230F39A842A54821FDD223092819D7728EFB9D3A0080B75380B"
    )
)
output = subprocess.check_output(f"echo {expected_detached_cose_message.dumps().hex().upper()} | pretty2diag.rb -e", shell=True).decode(sys.stdout.encoding).rstrip("\n")
print(output)
```

on the other hand,
## I expect that `countersign` on this instance would produce we don't expect
because both self._payload and self._msg[2] is `None` .

**Comment by dajiaji** [comment]
> Could you fix it in the future?

Yes, I understand this negative side-effect. I can fix it on my next PR.

**Review by dajiaji — APPROVED** [review]

**Pull request merged**